src/transport_ofi.c

```diff
@@ -297,7 +297,15 @@ static inline int allocate_endpoints(struct fabric_info *info)
      * support shmem_wait that blocks on an update arriving from the network */
 #ifndef ENABLE_HARD_POLLING
     info->p_info->caps |= FI_REMOTE_WRITE | FI_REMOTE_READ;
+#else
+    info->p_info->caps &= ~FI_REMOTE_WRITE;
+    info->p_info->caps &= ~FI_REMOTE_READ;
```

Comment:
Discussed with Kayla here. Adding permissions to read/write if not hard polling seems wrong to me. You need the permissions to read/write anyways (and already covered under FI_RMA), and hard polling only controls whether you look at the target counter or not e.g. FI_RMA_EVENT.